lib/product-import.coffee

```diff
@@ -230,6 +232,52 @@ class ProductImport
       else
         false
 
+  _updateProductRepeater: (prodToProcess, existingProduct) ->
+    repeater = new Repeater {attempts: 5}
+    repeater.execute =>
+      @_updateProduct(prodToProcess, existingProduct)
+    , (e) =>
+      if e.statusCode isnt 409 # concurrent modification
+        return Promise.reject e
+
+      @logger.warn "Recovering from 409 concurrentModification error on product '#{existingProduct.id}'"
+
+      Promise.resolve => # next task must be a function
+        @client.productProjections.staged(true).byId(existingProduct.id).fetch()
+        .then (result) =>
+          @_updateProduct(prodToProcess, result.body)
+
+  _updateProduct: (_prodToProcess, existingProduct) ->
+    prodToProcess = _.deepClone(_prodToProcess)
```

Comment:
It adds some `_SOMETHING` variables to `product` object and if you send there for example categories with their MongoDB ids (instead of externalIds) it will tell you that it was not able to resolve category reference and it will skip this product. Let me do something with it..